plenigo_plugin/PlenigoContentManager.php

```diff
@@ -684,15 +693,23 @@ private function replace_plenigo_tags($html) {
         $prodDetails = '*[ERROR check Product ID]';
         $courtTitle = $this->options['curtain_title'];
         $courtMsg = $this->options['curtain_text'];
-        $btnTitle = $this->options['curtain_buy'];
-        $btnOnClick = "javascript:alert('This product has not been configured correctly!');";
-        $btnStyle = "width:30%";
-        $loginTitle = $this->options['curtain_login'];
-        $loginOnClick = "javascript:alert('The login API doesn't work!');";
+        //Get buy text of the tag DB
+        $buyTitle = $this->get_buy_text(isset($this->options['curtain_buy']) ? $this->options['curtain_buy'] : '');
+        $buyOnClick = "javascript:alert('This product has not been configured correctly!');";
         $loginStyle = "width:30%";
-        $custTitle = $this->options['curtain_custom_title'];
-        $custOnClick = "javascript:window.location.href = '" . $this->options['curtain_custom_url'] . "';";
         $custStyle = "width:30%";
+        $buyStyle = "width:30%";
+        $loginTitle = $this->options['curtain_login'];
+        $loginOnClick = "javascript:alert('The login API doesn't work!');";
+        if (!isset($this->reqCache["lastCatId"])) {
+            $curtainMode = (isset($this->options['curtain_mode'])) ? $this->options['curtain_mode'] : 1;
```

Comment:
A constant for the curtain mode would be great. We are having this magic numbers 1,2,3,4 all over the code.